.github/workflows/ci.yml

```diff
@@ -187,7 +187,7 @@ jobs:
 
       - name: "Upload to codecov.io"
         if: ${{ contains(env['RUN_ANALYZER'], 'cov') }}
-        uses: codecov/codecov-action@d9f34f8cd5cb3b3eb79b3e4b5dae3a16df499a70 # pin@v3
+        uses: codecov/codecov-action@v4
```

Comment:
@kahest, we had this already once: https://github.com/getsentry/sentry-native/pull/767#discussion_r1000620544

Can you tell me how other (SDK) repos can use Dependabot to update between major versions while we have to pin on a SHA: https://github.com/getsentry/sentry-python/pull/2706/files? Also, it seems like in the `getsentry` org, most repos use `v3` or `v4`, but those that pin definitely do not all use the same SHA.

Especially now, since Codecov is part of Sentry, we should be able to trust them on their major update path, no? Is there any update on how to handle this?